src/MPAGSCipher/VigenereCipher.cpp

```diff
@@ -0,0 +1,72 @@
+#include "VigenereCipher.hpp"
+#include "Alphabet.hpp"
+#include "CipherMode.hpp"
+
+#include <algorithm>
+#include <iostream>
+
+VigenereCipher::VigenereCipher ( const std::string& key )
+{
+    // Set the given key
+    setKey(key);
+}
+
+void VigenereCipher::setKey(const std::string& key) {
+
+    // Store the original key
+    key_ = key;
+
+    // Make sure the key is uppercase
+    std::transform(std::begin(key_), std::end(key_), std::begin(key_), ::toupper);
+
+    // Remove non-alphabet characters
+    key_.erase(std::remove_if(std::begin(key_), std::end(key_), [](char c) { return !std::isalpha(c); }), std::end(key_));
+
+    // Check if the key is empty and replace with default if so
+    if (key_.empty()){
+        key_ = "MYKEY";
+    }
+    // loop over the key
+    for (char c : key_){
+
+        // Find the letter position in the alphabet
+        const size_t position = Alphabet::alphabet.find(c);
+
+        // Create a CaesarCipher using this position as a key
+        CaesarCipher caesarCipher(position);
+
+        // Insert a std::pair of the letter and CaesarCipher into the lookup
+        charLookup_.insert(std::make_pair(c, caesarCipher));
+    }
```

Comment:
This is all good.  In our solution we first check whether we've already done this letter by checking if it's in the map, but that isn't really necessary since any attempted repeat insertions will fail silently anyway.